File: src/codeforces/contest1731/ProblemB.java
package codeforces.contest1731;

import java.util.Scanner;

public class ProblemB {
    private static final int p = 1_000_000_007;

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int tests = scanner.nextInt();
        for (int t = 0; t < tests; t++) {
            int n = scanner.nextInt();
            System.out.println(solve(n));
        }
    }

    private static long solve(int n) {
        //1/6 (n-1) n (1 + 4 n)
        long ans = 1;
        boolean dividedByTwo = false, dividedByThree = false;
        long[] m = {n - 1, n, 4L * n + 1};
        for (long a : m) {
            if (!dividedByTwo && a % 2 == 0) {
                a /= 2;
                dividedByTwo = true;
            }
            if (!dividedByThree && a % 3 == 0) {
                a /= 3;
                dividedByThree = true;
            }
            ans *= a;
            if (ans >= p) ans %= p;
        }
        ans += (long) n * n;
        if (ans >= p) ans %= p;
        ans *= 2022;
        if (ans >= p) ans %= p;
        return (int) (ans);
    }
}

File: src/codeforces/contest1927/ProblemB.kt
@file:Suppress("DuplicatedCode")

package codeforces.contest1927

import java.io.BufferedReader
import java.io.InputStream
import java.io.InputStreamReader
import java.util.*

fun main(args: Array<String>) {
    val tests = nextInt()
    for (t in 0 until tests) println(solve(nextInts(nextInt())))
}

fun solve(arr: IntArray): String {
    val n = arr.size
    val cnt = IntArray(26)
    val ans = CharArray(n)
    for (i in 0 until n) {
        for (c in 0 until 26) {
            if (cnt[c] == arr[i]) {
                ans[i] = 'a' + c
                cnt[c]++
                break
            }
        }
    }
    return String(ans)
}


private class MyReader2(inputStream: InputStream) {
    private val reader = BufferedReader(InputStreamReader(inputStream))
    private var tokenizer: StringTokenizer? = null

    fun next(): String {
        while (true) {
            tokenizer.let {
                if (it == null || !it.hasMoreTokens()) {
                    tokenizer = StringTokenizer(reader.readLine())
                } else {
                    return it.nextToken()
                }
            }
        }
    }
}

private val reader = MyReader2(System.`in`)
private fun next() = reader.next()
private fun nextInt() = next().toInt()
private fun nextLong() = next().toLong()
private fun nextDouble() = next().toDouble()
private fun nextStrings(n: Int) = Array(n) { next() }
private fun nextInts(n: Int) = IntArray(n) { nextInt() }
private fun nextLongs(n: Int) = LongArray(n) { nextLong() }
private fun nextDoubles(n: Int) = DoubleArray(n) { nextDouble() }